builder/common/step_network_info_test.go

```diff
@@ -0,0 +1,409 @@
+				switch filterVal {
+				case "us-east-1a":
+					machines = []*ec2.InstanceTypeOffering{
+						{
+							InstanceType: aws.String("t2.mini"),
+						},
+						{
+							InstanceType: aws.String("t3.large"),
+						},
+					}
+				case "us-east-1b":
+					machines = []*ec2.InstanceTypeOffering{
+						{
+							InstanceType: aws.String("t2.mini"),
+						},
+						{
+							InstanceType: aws.String("t2.micro"),
+						},
+					}
+				default:
+					return nil, fmt.Errorf("error: location %q not expected", filterVal)
+				}
+			}
+
+			if !foundLocation {
+				return nil, fmt.Errorf("couldn't find location in filters")
+			}
+
+			return &ec2.DescribeInstanceTypeOfferingsOutput{
+				InstanceTypeOfferings: machines,
+			}, nil
+		},
+	}
+
+	stepConfig := &StepNetworkInfo{
+		AssociatePublicIpAddress: confighelper.TriTrue,
+		RequestedMachineType:     "t3.large",
+	}
+
+	state := &multistep.BasicStateBag{}
+	state.Put("ec2", mockConn)
+	state.Put("ui", &packersdk.MockUi{})
+
+	actRet := stepConfig.Run(context.Background(), state)
+	if actRet == multistep.ActionHalt {
+		t.Fatalf("running the step failed: %s", state.Get("error").(error))
+	}
+
+	vpcid, ok := state.GetOk("vpc_id")
+	if !ok || vpcid != "default-vpc" {
+		t.Errorf("error: vpc should be 'default-vpc', but is %q", vpcid)
+	}
+	t.Logf("set vpc is %q", vpcid)
+
+	subnetid, ok := state.GetOk("subnet_id")
+	if !ok || subnetid != "subnet1" {
+		t.Errorf("error: subnet should be 'subnet1', but is %q", subnetid)
+	}
+	t.Logf("set subnet is %q", subnetid)
+
+	az, ok := state.GetOk("availability_zone")
+	if !ok || az != "us-east-1a" {
+		t.Errorf("error: availability_zone should be 'us-east-1a', but is %q", az)
+	}
+	t.Logf("set AZ is %q", az)
+}
+
+func TestGetDefaultVPCFailDueToPermissions(t *testing.T) {
+	mockConn := &mockEC2ClientStepNetworkTests{
+		describeVpcs: func(dvi *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error) {
+			return nil, fmt.Errorf("Insufficient permissions: missing ec2:DescribeVpcs")
+		},
+	}
+
+	stepConfig := &StepNetworkInfo{
+		AssociatePublicIpAddress: confighelper.TriTrue,
+		RequestedMachineType:     "t3.large",
+	}
+
+	ui := &packersdk.MockUi{}
+
+	state := &multistep.BasicStateBag{}
+	state.Put("ec2", mockConn)
+	state.Put("ui", ui)
+
+	actRet := stepConfig.Run(context.Background(), state)
+	if actRet == multistep.ActionHalt {
+		t.Fatalf("running the step failed: %s", state.Get("error").(error))
+	}
+
+	vpcid, ok := state.GetOk("vpc_id")
+	if !ok || vpcid != "" {
+		t.Errorf("error: vpc should be empty, but is %q", vpcid)
+	}
+	t.Logf("set vpc is %q", vpcid)
+
+	subnetid, ok := state.GetOk("subnet_id")
+	if !ok || subnetid != "" {
+		t.Errorf("error: subnet should be empty, but is %q", subnetid)
+	}
+	t.Logf("set subnet is %q", subnetid)
+
+	az, ok := state.GetOk("availability_zone")
+	if !ok || az != "" {
+		t.Errorf("error: availability_zone should be empty, but is %q", az)
+	}
+	t.Logf("set AZ is %q", az)
+
+	ok = false
```

Comment:
Since `ok` has already been declared before I have to set it to false, but alternatively we can declare a new flag for use in the loop below. Any preference?

Reply:
Oh I see. You're are reusing the ok from the GetOk call. Yeah that is confusing. In fact this is a good place to use the if ok convention. Which is probably what my. mind read by default. In this particular case since the ok value is going to be used for another unrelated test I would suggest rewriting the GetOk check or renaming this variable to found.

Reply:
I opted to rename this bool for the check